### PR TITLE
Pass enrollmentId in ADAL when saving broker response

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: ADAL/src/request/ADAuthenticationRequest+Broker.m [hunk 1]
```diff
@@ -310,9 +312,19 @@ + (ADAuthenticationResult *)processBrokerResponse:(NSURL *)response
...
         MSIDDefaultTokenCacheAccessor *otherAccessor = [[MSIDDefaultTokenCacheAccessor alloc] initWithDataSource:dataSource otherCacheAccessors:nil factory:factory];
         MSIDLegacyTokenCacheAccessor *cache = [[MSIDLegacyTokenCacheAccessor alloc] initWithDataSource:dataSource otherCacheAccessors:@[otherAccessor] factory:factory];
+        
+        NSString *userId = [[[result tokenCacheItem] userInformation] userId];
+        NSString *applicationidentifier = [ADRequestParameters applicationIdentifierWithAuthority:brokerResponse.authority];
+        NSString *enrollmentId = nil;
+        
+        if (applicationidentifier)
+        {
+            enrollmentId = [ADEnrollmentGateway enrollmentIDForHomeAccountId:nil userID:userId error:nil];
```
Comment (jasoncoolmax) [inline ADAL/src/request/ADAuthenticationRequest+Broker.m]: do we want to provide home account here as well?

File: ADAL/src/request/ADAuthenticationRequest+Broker.m [hunk 2]
```diff
@@ -310,9 +312,19 @@ + (ADAuthenticationResult *)processBrokerResponse:(NSURL *)response
         MSIDOauth2Factory *factory = [MSIDAADV1Oauth2Factory new];
         MSIDDefaultTokenCacheAccessor *otherAccessor = [[MSIDDefaultTokenCacheAccessor alloc] initWithDataSource:dataSource otherCacheAccessors:nil factory:factory];
         MSIDLegacyTokenCacheAccessor *cache = [[MSIDLegacyTokenCacheAccessor alloc] initWithDataSource:dataSource otherCacheAccessors:@[otherAccessor] factory:factory];
+        
+        NSString *userId = [[[result tokenCacheItem] userInformation] userId];
```
Comment (jasoncoolmax) [inline ADAL/src/request/ADAuthenticationRequest+Broker.m]: Do we want to use the dot notation rather than brackets here to make it easier to understand :)

Comment (antrix1989) [reply ADAL/src/request/ADAuthenticationRequest+Broker.m]: +1 :)

File: ADAL/src/request/ADAuthenticationRequest+Broker.m (hunk 1 above)
Comment (oldalton) [reply ADAL/src/request/ADAuthenticationRequest+Broker.m]: addressed :)

File: ADAL/src/request/ADAuthenticationRequest+Broker.m (hunk 2 above)
Comment (oldalton) [reply ADAL/src/request/ADAuthenticationRequest+Broker.m]: addressed :)

**Pull request merged**